src/index.js

```diff
@@ -9,21 +9,23 @@ function yieldTypes(schema) {
   return schema.types;
 }
 
-function filterTypes(whitelist) {
+function filterTypes(whitelistConfig) {
   return function(types) {
-    if (!whitelist) {
+    if (!whitelistConfig) {
       return types;
     }
 
+    const whitelistTypes = Object.getOwnPropertyNames(whitelistConfig);
```

Comment:
`keys` should act the same. ie: own properties only. Only difference is it doesn't return non-enumerable properties. `for .. in ` is the method that also has properties from the prototype chain (according to MDN).